src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3639,6 +3649,710 @@ .maxstack  3
+                );
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_WithParamsCopyConstructor()
+        {
+            var source =
+@"public record B(object N1, object N2)
+{
+    public B(B b, params int[] i) : this(30, 40) { }
+}
+public record C(object P1, object P2) : B(3, 4) { }
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var actualMembers = comp.GetMember<NamedTypeSymbol>("B").GetMembers().Where(m => m.Name == ".ctor").ToTestDisplayStrings();
+            var expectedMembers = new[]
+            {
+                "B..ctor(System.Object N1, System.Object N2)",
+                "B..ctor(B b, params System.Int32[] i)",
+                "B..ctor(B )"
+            };
+            AssertEx.Equal(expectedMembers, actualMembers);
+
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       32 (0x20)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""B..ctor(B)""
+  IL_0007:  ldarg.0
+  IL_0008:  ldarg.1
+  IL_0009:  ldfld      ""object C.<P1>k__BackingField""
+  IL_000e:  stfld      ""object C.<P1>k__BackingField""
+  IL_0013:  ldarg.0
+  IL_0014:  ldarg.1
+  IL_0015:  ldfld      ""object C.<P2>k__BackingField""
+  IL_001a:  stfld      ""object C.<P2>k__BackingField""
+  IL_001f:  ret
+}
+");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_WithInitializers()
+        {
+            var source =
+@"public record C(object N1, object N2)
+{
+    private int field = 42;
+    public int Property = 43;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       55 (0x37)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""object..ctor()""
+  IL_0006:  ldarg.0
+  IL_0007:  ldarg.1
+  IL_0008:  ldfld      ""object C.<N1>k__BackingField""
+  IL_000d:  stfld      ""object C.<N1>k__BackingField""
+  IL_0012:  ldarg.0
+  IL_0013:  ldarg.1
+  IL_0014:  ldfld      ""object C.<N2>k__BackingField""
+  IL_0019:  stfld      ""object C.<N2>k__BackingField""
+  IL_001e:  ldarg.0
+  IL_001f:  ldarg.1
+  IL_0020:  ldfld      ""int C.field""
+  IL_0025:  stfld      ""int C.field""
+  IL_002a:  ldarg.0
+  IL_002b:  ldarg.1
+  IL_002c:  ldfld      ""int C.Property""
+  IL_0031:  stfld      ""int C.Property""
+  IL_0036:  ret
+}");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor()
```

Comment:
>CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor [](start = 20, length = 52)

It is quite possible I missed the tests, but it looks like we are not testing user-defined copy constructors in the top level record with:
- no initializer
- this initializer
- base initializer

It would also be good to add test covering user-defined copy ctor without initializer and synthesized copy ctor when System.Object lacks parameter-less constructor.

Reply:
Adding the first three with explicit verification.

For bad object type, I didn't touch that code and have a time constraint. So will skip that.

---
In reply to: [438514907](https://github.com/dotnet/roslyn/pull/45022#discussion_r438514907) [](ancestors = 438514907)